agent/setup.go

```diff
@@ -0,0 +1,168 @@
+package agent
```

Comment:
I fully support splitting these out into a setup file - the delineation makes sense.